tests/Functional/Service/Signalement/QualificationStatusServiceTest.php

```diff
@@ -38,4 +117,120 @@ public function testUpdateNonDecenceEnergetiqueStatus()
+
+        $status = $qualificationService->getNDEStatus($signalementQualification);
+
+        $this->assertEquals(QualificationStatus::NDE_OK, $status);
+    }
+
+    public function testUpdateNdeStatusDpeAfter2023Averee()
+    {
+        $signalementRepository = $this->entityManager->getRepository(Signalement::class);
+
+        /** @var Signalement $signalement */
+        $signalement = $signalementRepository->findOneBy(['reference' => '2023-8']);
+        /** @var SignalementQualification $signalementQualification */
+        $signalementQualification = $signalement->getSignalementQualifications()[0];
+
+        $qualificationNDERequest = new QualificationNDERequest(
+            dateEntree: '2023-02-01',
+            dateDernierBail: '2023-02-01',
+            dateDernierDPE: '2023-02-01',
+            superficie: 100,
+            consommationEnergie: 580,
+            dpe: true
+        );
+        $signalementQualification->setDernierBailAt(new DateTimeImmutable($qualificationNDERequest->getDateDernierBail()));
+        $signalementQualification->setDetails($qualificationNDERequest->getDetails());
+        $signalement->setSuperficie($qualificationNDERequest->getSuperficie());
+
+        $qualificationService = new QualificationStatusService();
+        $signalementQualification->setStatus($qualificationService->getNDEStatus($signalementQualification));
+
+        $status = $qualificationService->getNDEStatus($signalementQualification);
+
+        $this->assertEquals(QualificationStatus::NDE_AVEREE, $status);
+    }
+
+    public function testUpdateNdeStatusDpeAfter2023Ok()
+    {
+        $signalementRepository = $this->entityManager->getRepository(Signalement::class);
+
+        /** @var Signalement $signalement */
+        $signalement = $signalementRepository->findOneBy(['reference' => '2023-8']);
+        /** @var SignalementQualification $signalementQualification */
+        $signalementQualification = $signalement->getSignalementQualifications()[0];
+
+        $qualificationNDERequest = new QualificationNDERequest(
+            dateEntree: '2023-02-01',
+            dateDernierBail: '2023-02-01',
+            dateDernierDPE: '2023-02-01',
+            superficie: 100,
+            consommationEnergie: 320,
+            dpe: true
+        );
+        $signalementQualification->setDernierBailAt(new DateTimeImmutable($qualificationNDERequest->getDateDernierBail()));
+        $signalementQualification->setDetails($qualificationNDERequest->getDetails());
+        $signalement->setSuperficie($qualificationNDERequest->getSuperficie());
+
+        $qualificationService = new QualificationStatusService();
+        $signalementQualification->setStatus($qualificationService->getNDEStatus($signalementQualification));
+
+        $status = $qualificationService->getNDEStatus($signalementQualification);
+
+        $this->assertEquals(QualificationStatus::NDE_OK, $status);
+    }
```

Comment:
Il y'a une feature top pour factoriser des tests sur phpunit que tu peux utiliser dans ton cas ce sont les dataprovider

https://docs.phpunit.de/en/10.0/writing-tests-for-phpunit.html#writing-tests-for-phpunit-data-providers-examples-numericdatasetstest-php

Tu trouveras pas mal d'exemples sur le projet en faisant une recherche sur `@dataProvider`

Reply:
Génial ! C'est fait ! Merci :)